WooCommerce/Classes/ViewRelated/Dashboard/Onboarding/StoreOnboardingViewModel.swift

```diff
@@ -161,6 +175,27 @@ private extension StoreOnboardingViewModel {
         defaults[.completedAllStoreOnboardingTasks] = true
     }
 
+    @MainActor
+    func isFreeTrialPlan() async -> Bool {
+        // Only fetch free trial information if the site is a WPCom site.
+        guard stores.sessionManager.defaultSite?.isWordPressComStore == true else {
+            return false
+        }
+
+        return await withCheckedContinuation({ continuation in
+            let action = PaymentAction.loadSiteCurrentPlan(siteID: siteID) { result in
+                switch result {
+                case .success(let plan):
+                    return continuation.resume(returning: plan.isFreeTrial)
+                case .failure(let error):
+                    DDLogError("⛔️ Error fetching the current site's plan information: \(error)")
+                    continuation.resume(returning: false)
```

Comment:
nit: it might look more consistent with the success case also returning here
```suggestion
                    return continuation.resume(returning: false)
```

Reply:
Done in 5a349b2aca5fd43876f28eaef4696251c72706d5